src/extension/actor.ts

```diff
@@ -57,12 +77,15 @@ export function createActor<
     }
   }
 
-  const on: Actor<Messages>["on"] = (name, callback) => {
+  const on: Actor["on"] = (name, callback) => {
     let listeners = messageListeners.get(name) as Set<typeof callback>;
 
     if (!listeners) {
       listeners = new Set();
-      messageListeners.set(name, listeners as Set<(message: Messages) => void>);
+      messageListeners.set(
+        name,
+        listeners as Set<(message: ActorMessage) => void>
+      );
     }
```

Comment:
```suggestion
    if (!listeners) {
      listeners = new Set();
      messageListeners.set(name, listeners);
    }
```

Seems the cast is not neccessary here

Reply:
Good catch!